CodeStarterCamp_Week3/Round3Step2/CoffeeShopStep2.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  CoffeeShopStep2.swift
+//  CodeStarterCamp_Week3
+//
+//  Created by QL Jeong on 2023/08/31.
+//
+
+import Foundation
+
+class CoffeeShopStep2: CoffeeShop {
+    func getPrice(of coffee: Coffee) -> Int {
+        guard let item = getMenuItem(of: coffee) else {
+            return -1
+        }
+
+        return item.price
+    }
```

Comment:
요 부분에서 item을 가져오지 못했을 때 -1을 반환하는 의미를 직관적으로 알기 어려운 것 같습니다.
`Error`를 활용하면 잘 해결되겠네요!

Error 프로토콜을 채택한 열거형의 case는 `throws` 키워드가 있는 함수 / 메서드에서 반환값 대신 오류를 던져줄 수 있습니다.
던져진 오류를 처리하는 방법에는 do - catch문, Result타입 활용이 있습니다.
관련 내용은 차주 학습자료를 통해 학습하시면 좋겠습니다 👍 

```suggestion
    func getPrice(of coffee: Coffee) throws -> Int {
        guard let item = getMenuItem(of: coffee) else {
            throw OrderError.isNotInMenu
        }

        return item.price
    }
```

Reply:
아직 누적 시간이 부족하여 다음 과정이 열리지 않아서 어떤 내용인지 확인 하지는 못하였습니다.
일단은 제가 아는 방법으로 구현 하였습니다.
말씀 대로 다음 프로젝트에서 지적 하실부분을 말씀 해 주시면 되겠습니다.

---

throws 로 Error 를 반환 한다고 선언 하면
catch 쪽에서 해당 프로젝트 또는 모듈에서 사용 가능한 모든 Error 가 대상이 되어서
부분적으로만 처리를 하고 싶을때에는 약간은 귀찮은 부분이 생겨 나서 조금은 머뭇 거리게 됩니다.
물론 발생할 수 있는 모든 Error 가 나오게 됨으로 종류별로 각각 후처리가 필요한 경우에는 사용 합니다.

저는 method 에서 Error 를 단한가지만 발생 하는것이 아니거나
명시적으로 후처리가 필요한경우 가능한 Result 로 처리 하고 있긴 합니다.

이 경우 반환 하는 내용은 음수 로는 "없다" 를 뜻하고
0 을 포함한 양수 는 "있다" 라는 내용으로서 예외 처리는 1개의 경우라 이렇게 표현 하였습니다.

만약 음수값에 대한 처리가 알아보기 어렵다면 Error 보다는 아래처럼 optional 로 바꾸어도 될꺼라 생각 합니다.
```swift
func getPrice(of coffee: Coffee) -> Int? {
  // 해당하는 메뉴가 없으면 nil
  // 있다면 가격 반환
}
```

Reply:
메뉴가 없을 때 nil을 반환해주는 편이 더 좋긴 하겠네요!! 👍

Reply:
적용 하였습니다.